app/main.py

```diff
@@ -302,28 +303,34 @@ def grab_secrets(self):
                     cert_file.write(self._cert_to_pem(cert.cer))
     
     def _load_cert(self, secret_value, content_type):
+        x509 = ()
         if (content_type == 'application/x-pkcs12'):
             pkcs12 = crypto.load_pkcs12(base64.decodestring(secret_value))
-            x509 = pkcs12.get_certificate()
+            ca_certs = (pkcs12.get_ca_certificates() or ())
+            cert = pkcs12.get_certificate()
+            x509 += (cert,) + ca_certs
             key = pkcs12.get_privatekey()
         elif (content_type == 'application/x-pem-file'):
-            x509 = crypto.load_certificate(crypto.FILETYPE_PEM, secret_value.encode('utf-8'))
+            start_line = b'-----BEGIN CERTIFICATE-----'
+            cert_slots = secret_value.split(start_line)
+            for single_pem_cert in cert_slots[1:]:
+                cert = crypto.load_certificate(crypto.FILETYPE_PEM, start_line+single_pem_cert)
+                x509 += (cert,)
             # extract the private key portion of PEM
-            key = crypto.load_privatekey(crypto.FILETYPE_PEM, secret_value.encode('utf-8'))
-            pkcs12 = None
+            key = crypto.load_privatekey(crypto.FILETYPE_PEM, cert_slots[0])
```

Comment:
This is assuming that the private key always is at the top of the PEM.  I am unsure if that is the requirement.

Just pass secret_value here rather than cert_slots[0] since keys are not cert types but key types.  Plus, it will find the first (and usually only) key in the PEM file this way.